Makefile

```diff
@@ -194,32 +193,64 @@ $(TERRAFORM):
 	@rm -fr $(TOOLS_HOST_DIR)/tmp-terraform
 	@$(OK) installing terraform $(HOSTOS)-$(HOSTARCH)
 
-$(TERRAFORM_PROVIDER):
-	@$(INFO) installing terraform AWS provider for the platform $(SAFEHOST_PLATFORM)
-	@mkdir -p $(dir $(TERRAFORM_PROVIDER))
-	@curl -fsSL https://github.com/upbound/terraform-provider-aws/releases/download/$(TERRAFORM_PROVIDER_RELEASE)/terraform-provider-aws_$(TERRAFORM_PROVIDER_RELEASE)_$(SAFEHOST_PLATFORM).zip -o $(TERRAFORM_PROVIDER).zip
-	@unzip $(TERRAFORM_PROVIDER).zip -d $(dir $(TERRAFORM_PROVIDER))
-	@rm -fr $(TERRAFORM_PROVIDER).zip
-	@$(OK) installing terraform provider $(SAFEHOST_PLATFORM)
+$(TERRAFORM_PROVIDER): $(UPSTREAM)
+	@$(INFO) Building terraform AWS provider from source if changes detected
+	@(cd $(ROOT_DIR)/upstream && make go-build)
+	@$(OK) Built terraform AWS provider from source
 
+# TODO(mbbush) Remove this when we can. This is the last thing left over from when we used to fork a terraform cli
+# process for each resource. This generates the tf provider's json schema, which is slightly different from the go
```

Comment:
As I've dug deeper into the various schemas, I'm less sure that we should remove this entirely, so I'll update the comment.